### DISCUSSION TITLE
How to start machine from nested state?

### DISCUSSION THREAD

**Discussion by bennyKY** [statelyai/xstate#2063] [Q&A] [answered]
Hi!
Having this machine:
```javascript
const machine = Machine({
  initial: 'off',
  states: {
    off: { on: { CLICK: 'on' } },
    on: {
      on: { CLICK: 'off' },
      initial: 'red',
      states: {
        red: { on: { STEP: 'yellow' } },
        yellow: { on: { STEP: 'green' } },
        green: { on: { STEP: 'red' } },
      },
    },
  },
```

in order to start it: ```interpret(machine).start()```
this can work as well: ```interpret(machine).start('on)``` or ```interpret(machine).start('off')```.
**Is there a way to start the machine from a nested state**? 
i.e something like: ```interpret(machine).start('on.red')``` (it doesn't work)

**Comment by RainerAtSpirit** [comment]
Consider using a factory.

```javascript
function machineFactory(initial = "off", initialSub = "red") {
  const machine = Machine({
    initial,
    states: {
      off: { on: { CLICK: "on" } },
      on: {
        initial: initialSub,
        states: {
          red: { on: { STEP: "yellow" } },
          yellow: { on: { STEP: "green" } },
          green: { on: { STEP: "red" } }
        }
      }
    },
    on: { CLICK: "off" }
  });
  return machine;
}

const myMachine = machineFactory("on", "yellow");

https://codesandbox.io/s/boring-field-de5jl?file=/src/index.js

**Reply by bennyKY** [reply]
@RainerAtSpirit Thanks!
Looks good, but i just wondered if there is "official" supported way to do it. In my experience with xstate, if its not supported - than im doing something wrong 😆

**Comment by Andarist** [comment, marked as answer]
I would have to recheck the implementation and all to be sure but i think that this might work:
```js
service.start({ on: 'red' })
```

**Reply by bennyKY** [reply]
Works. Awesome! Thanks!

**Reply by bennyKY** [reply]
@Andarist i would suggest to add it to the docs
